src/platform/backends/shared/base_virtual_machine.h

```diff
@@ -129,15 +130,22 @@ class BaseVirtualMachine : public VirtualMachine
 
 } // namespace multipass
 
-inline int multipass::BaseVirtualMachine::get_num_snapshots() const noexcept
+inline int multipass::BaseVirtualMachine::get_num_snapshots() const
 {
+    require_snapshots_support();
     return static_cast<int>(snapshots.size());
 }
 
 inline int multipass::BaseVirtualMachine::get_snapshot_count() const
 {
+    require_snapshots_support();
     const std::unique_lock lock{snapshot_mutex};
     return snapshot_count;
 }
 
+inline void multipass::BaseVirtualMachine::require_snapshots_support() const
+{
+    throw NotImplementedOnThisBackendException{"Snapshots"};
```

Comment:
I think this should be lower case `snapshots`.  Making it lower case is also in line with `networks` and the like.

Reply:
Good point. Done.